pkg/rtc/subscribedtrack.go

```diff
@@ -47,17 +48,19 @@ type SubscribedTrackParams struct {
 
 type SubscribedTrack struct {
 	params           SubscribedTrackParams
-	subMuted         atomic.Bool
-	pubMuted         atomic.Bool
-	settings         atomic.Pointer[livekit.UpdateTrackSettings]
 	logger           logger.Logger
 	sender           atomic.Pointer[webrtc.RTPSender]
 	needsNegotiation atomic.Bool
 
+	settingsLock sync.Mutex
```

Comment:
removing atomics and using a lock. There are three places where this can get called from which made atomics a bit scary
1. Signal - update track settings
2. Signal - update video layers. - this is not used and it also comes via signal channel which is effectively same as above
3. Debounced handler - to avoid frequent changes
4. On bind - to handle the tricky interplay for adaptive streaming/visibility/dynacast

So, bring all of it under a lock. This does cause some callbacks with lock held. Wll point out in code below. Open to better suggestions to avoid doing callback under lock.

Reply:
Actually, cases (1) and (2) are not effectively the same as one is from publisher signal channel and the other is from subscriber signal channel. Another  path via publish signal channel is publisher side mute. So, there are four distinct  paths where this gets applied.

Reply:
instead of calling the callbacks under lock we can generate a timed version synced to the callback arguments under lock and check the version in the callback handler

Reply:
Awesome, will have a look at it. Thank you @paulwe